base/env.jl

```diff
@@ -244,3 +244,30 @@ function withenv(f, keyvals::Pair{T}...) where T<:AbstractString
     end
 end
 withenv(f) = f() # handle empty keyvals case; see #10853
+
+"""
+    username() -> String
+
+Return the username for the current user. If the username cannot be found,
+this function throws an error.
+
+To retrieve a username that is overridable via an environment variable,
+e.g., `USER`, consider using
+```julia
+user = get(ENV, "USER", username())
+```
+
+!!! note
+    The username is determined via the user account and password database
+    that is available only on Unix-like platforms.
+
+!!! compat "Julia 1.11"
+    This function requires at least Julia 1.11.
+"""
+function username()
+    uid = Libc.getuid()
+    username = Filesystem.getusername(uid)
+    isnothing(username) || return username
+    # the user was not found or the username was empty
+    throw(ErrorException("username not found in the password database"))
```

Comment:
Is there a more suitable exception type?

Reply:
The system usually reports ENOENT for it as an IOError